apps/designer/app/canvas/shared/design-tokens.ts

```diff
@@ -0,0 +1,17 @@
+import type { DesignToken } from "@webstudio-is/design-tokens";
+import { useEffect, useState } from "react";
+import { useDesignTokens } from "~/shared/nano-states";
+import { publish } from "~/shared/pubsub";
+
+export const useInitializeDesignTokens = (designTokens: Array<DesignToken>) => {
+  const [, setDesignTokens] = useDesignTokens();
+  // Trick to set it only once.
+  useState(() => setDesignTokens(designTokens));
```

Comment:
Btw. In other places where you've replaced `useEffect` with `useState` (breakpoints.ts etc.), it will run on server now, but is it ok? I think we've started to use `useEffect` specifically because we didn't want it to run on the sever as it would write into a shared global variable

Reply:
Sorry, misread the diff. Nowhere, my mistake.